docs/architecture.md

```diff
@@ -102,33 +102,10 @@ Runtime supports the following auths
 
 To know how to add a new auth, see this doc {@tutorial new-auth-mechanisms}.
 
-Each auth has four hooks
+Each auth implements the 4 hooks in {@link AuthHandlerInterface}
```

Comment:
Implements `AuthHandlerInterface`?

Reply:
this line is followed by the 4 hooks, somehow need that in the sentence 
how about
```
Each auth implements the {@link AuthHandlerInterface} which has the following 4 hooks:
```